src/Components/App/App.tsx

```diff
@@ -14,36 +14,46 @@ function App() {
         email: "",
         password: "",
     });
-    // const [flipCardToIDE, setFlipCardToIDE] = useState<boolean>(false);
     const [showQuestionList, setShowQuestionList] = useState<boolean>(false);
     const navigate = useNavigate();
+    const [questions] = useLocalStorage(StorageKey.questionBank, []);
 
-    useEffect(() => {
-        const redirectUser = async () => {
-            if (userInfo.email) {
-                return navigate("/home");
-            } else {
-                return navigate("/login");
-            }
-        };
-        redirectUser();
+    const [currentQuestion, setCurrentQuestion] = useLocalStorage(
+        StorageKey.currentQuestion,
+        {}
+    );
+
+    const redirectUserQuestions = async () => {
+        if (showQuestionList && userInfo.email) {
+            return navigate("/question-list");
+        }
+    };
 
-        const redirectUserQuestions = async () => {
-            if (showQuestionList && userInfo.email) {
-                return navigate("/question-list");
-            }
-        };
+    const redirectUserFlashCard = async () => {
+        if (userInfo.email) {
+            return navigate("/flash-card");
+        } else {
+            return navigate("/login");
+        }
+    };
+
+    useEffect(() => {
+        redirectUserFlashCard();
         redirectUserQuestions();
     }, [userInfo, showQuestionList]);
 
     return (
         <div className="App">
-            <NavBar />
+            <NavBar
+                setShowQuestionList={setShowQuestionList}
+                redirectUserQuestions={redirectUserQuestions}
+                redirectUserFlashCard={redirectUserFlashCard}
```

Comment:
🤮 Don't pass state around this way in your nav. Nav will just navigate you to different pages. The logic needs to be contained in the separate pages